platform/stm32f0xx/platform.mk

```diff
@@ -44,6 +44,7 @@ endif
 DEFAULT_LINKER_SCRIPT ?= stm32f0_default.ld
 ifeq ($(MAKECMDGOALS),temp-bootloader-write)
 DEFAULT_LINKER_SCRIPT := stm32f0_application.ld
+CFLAGS += -DAPP_COMP_ON_BOOT
```

Comment:
What does `APP_COMP_ON_BOOT` stand for? `BOOTLOADER_APPLICATION` might be clearer.